lib/user.rb

```diff
@@ -0,0 +1,59 @@
+require 'bcrypt'
+
+class User
+
+  def self.new_user(username, password, email)
+    return 1 unless password.length >= 8
+    return 2 if duplicate_username?(username) == true 
+    return 3 if duplicate_email?(email) == true
+
+    encrpyted_pw = BCrypt::Password.create(password)
+
+    new_user = DbConnection.query("INSERT INTO users(username, password, email) 
+                                   VALUES ('#{username}', '#{encrpyted_pw}', 
+                                   '#{email}') RETURNING user_id, username, email;")
+    User.new(user_id: new_user[0]['user_id'], username: new_user[0]['username'], 
+             email: new_user[0]['email'])
+  end
+
+  def self.find(user_id)
+    return nil unless user_id
+
+    found = DbConnection.query("SELECT * FROM users WHERE user_id = '#{user_id}';")
+    User.new(user_id: found[0]['user_id'], username: found[0]['username'], email: found[0]['email'])
+  end
+
+  def self.find_by_username(username)
```

Comment:
could there be just one find method where you use the id to find the user details?